### PR TITLE
Introduces multicast announce interval

### PR DESCRIPTION
**Pull request by eelcocramer**
Multicast announcements repeat every X seconds. X defaults to 10. Repetition can be disabled by setting X to 0.

### CODE REVIEW

**Comment by clockworksoul** [comment]
Greetings, Eelcocramer! Thanks for this (and for fixing some of my various other commenting errors).

My concern with this is that with larger clusters this (and with multicast responses in general) could result in massive amounts of traffic as all receiving nodes respond simultaneously, essentially DDOS-ing the node and maybe the network.

I think that the best solution would be to remove the logic for the receiving node to respond _at all_. That way, the receiver will simply add the announcer to its registry, and will ping it at the normal point in its loop through known nodes. This provides strong knowledge consistence without taking anything down. 😄 

I was going to make that change but I think it would work nicely here. What do you think?

File: membership.go
Comment (clockworksoul) [inline membership.go]: Just a nitpick: could you add a mention of the loops behavior in the function comments?

File: properties.go
Comment (clockworksoul) [inline properties.go]: Comments should mention what time units are being used. The rest of the time values use milliseconds. Perhaps this should even use millis for consistency? I'm torn on that one... I'll leave it up to you.

File: properties.go
Comment (clockworksoul) [inline properties.go]: For consistency and clarity, this should include the units used as well: `DefaultMulticastAnnounceSeconds` or `DefaultMulticastAnnounceMillis` probably.

**Comment by eelcocramer** [comment]
Thanks for the comments! I will process them and make the necessary changes.

I agree that we can do without the response, I will try to add that to the PR.

Maybe the default behaviour should be _not_ to repeat the announcement?

**Comment by clockworksoul** [comment]
>I agree that we can do without the response, I will try to add that to the PR.

If you feel like it's too much of a change, we can break it into its own PR. :)

>Maybe the default behaviour should be not to repeat the announcement?

That's probably best, really. The announcement should only be necessary on startup, excepting some unusual use-cases. Once the new node is known to the cluster at large the SWIM protocol should be sufficient.

Perhaps if the network is given to extended partitioning, but Smudge's logic to occasionally retry "dead nodes" makes this less of an issue.

**Comment by eelcocramer** [comment]
> Perhaps if the network is given to extended partitioning, but Smudge's logic to occasionally retry "dead nodes" makes this less of an issue.

True, I am thinking about the case where nodes start of in different partition and then merge together but I agree this is not a primary use-case. I will change the default to 0.

File: membership.go
Comment (clockworksoul) [inline membership.go]: You probably don't mean to hard-code 10 here.

File: properties.go
Comment (clockworksoul) [inline properties.go]: Can this be changed to `EnvVarMulticastAnnounceIntervalSeconds` to agree with `DefaultMulticastAnnounceIntervalSeconds`?

File: properties.go
Comment (clockworksoul) [inline properties.go]: Can `DefaultMulticastEnabled` be moved up so that it follows its corresponding `EnvVar`?

**Review by clockworksoul — COMMENTED** [review]
Added a couple of nitpicks and a bug :)

File: membership.go
Comment (eelcocramer) [reply membership.go]: Good find 👍

**Review by clockworksoul — APPROVED** [review]
LGTM!

Would you like to merge, or shall I?

**Comment by eelcocramer** [comment]
Well, I do not have a merge button so if you're not merging it probably won't happen 😅

**Comment by clockworksoul** [comment]
Doh. Done and done! Thanks again!

**Pull request merged**
